safe_autonomy_sims/gym/inspection/sixdof_inspection_v0.py

```diff
@@ -467,14 +467,11 @@ def _get_reward(self):
         )
 
         # Sparse rewards
-        reward += (
-            r.weighted_inspection_success_reward(
-                chief=self.chief, total_weight=self.success_threshold
-            )
-            if utils.closest_fft_distance(chief=self.chief, deputy=self.deputy)
-            < self.crash_radius
-            else -1.0
-        )
+        success_reward = r.weighted_inspection_success_reward(chief=self.chief, total_weight=self.success_threshold)
+        if success_reward > 0:
+            if utils.closest_fft_distance(chief=self.chief, deputy=self.deputy) < self.crash_radius:
```

Comment:
We can combine these two if statements

Reply:
I originally had them combined but I don't think python was short-circuiting the statement properly. I got the biggest performance boost by separating them which made sure the fft function only ran when the success reward was positive.

Reply:
Interesting.  I'm going to play around with that a little bit to see what could have caused that

Reply:
If you can't reproduce it then I'm fine combining the statements. I got about a 25% speedup by splitting the statement. Not sure why since I believe python supports short-circuiting.

Reply:
I did a quick test to verify and Python definitely does support short-circuiting.  Initial thought is that JAX is somehow involved with the speedup.

Reply:
I'm not seeing a speedup in the pettingzoo environment between the two styles.  For Gym, I'm not seeing a speedup with the weighted inspection.  6DOF is giving me trouble with sometimes hanging or having a 0 norm quaternion error.  I've been using the sanity check scripts to get an idea of the performance.

Reply:
Yeah I'm not seeing the speedup anymore. Strange. May have just been a quirk on my end.

Reply:
I'm also getting issues with 6DOF. I haven't had a zero norm quat error yet but it's definitely hanging. It seems to be the ODE solver. It's adaptive time step selector is selecting a tiny step value (on the order of 1e-15) while solving and not correct back up to a reasonable step value. I'm not seeing why this is happening.

Reply:
I combined the if statements. Still looking into the 6dof issues.